example_test.go

```diff
@@ -0,0 +1,338 @@
+package sqlair_test
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/canonical/sqlair"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+type Location struct {
+	ID   int    `db:"room_id"`
+	Name string `db:"room_name"`
+}
+
+type Employee struct {
+	ID     int    `db:"id"`
+	TeamID int    `db:"team_id"`
+	Name   string `db:"name"`
+}
+
+type Team struct {
+	ID     int    `db:"id"`
+	RoomID int    `db:"room_id"`
+	Name   string `db:"team_name"`
+}
+
+func Example() {
+	sqldb, err := sql.Open("sqlite3", "file:example.db?cache=shared&mode=memory")
+	if err != nil {
+		panic(err)
+	}
+
+	db := sqlair.NewDB(sqldb)
+	createLocations := sqlair.MustPrepare(`
+	CREATE TABLE locations (
+		room_id integer,
+		room_name text
+	);`)
+
+	createEmployees := sqlair.MustPrepare(`
+	CREATE TABLE employees (
+		id integer,
+		team_id integer,
+		name text
+	);`)
+	createTeams := sqlair.MustPrepare(`
+	CREATE TABLE teams (
+		id integer,
+		room_id integer,
+		team_name text
+	)`)
+	createTables := []*sqlair.Statement{createLocations, createEmployees, createTeams}
+	for _, createTable := range createTables {
+		err = db.Query(nil, createTable).Run()
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	// Statement to populate the locations table.
+	insertLocation := sqlair.MustPrepare(`
+		INSERT INTO locations (room_name, room_id) 
+		VALUES ($Location.room_name, $Location.room_id)`,
+		Location{},
+	)
+
+	var l1 = Location{ID: 1, Name: "The Basement"}
+	var l2 = Location{ID: 2, Name: "Court"}
+	var l3 = Location{ID: 3, Name: "The Market"}
+	var l4 = Location{ID: 4, Name: "The Bar"}
+	var l5 = Location{ID: 5, Name: "The Penthouse"}
+	var locations = []Location{l1, l2, l3, l4, l5}
```

Comment:
[Nitpick] Prefer ` := ` initialization instead of `var`.

Reply:
Changed.